aspnetcore/fundamentals/routing.md

```diff
@@ -121,6 +121,13 @@ The methods provided by <xref:Microsoft.AspNetCore.Routing.LinkGenerator> suppor
 >
 > * Use <xref:Microsoft.AspNetCore.Routing.LinkGenerator> with caution in middleware in combination with `Map` or `MapWhen`. `Map*` changes the base path of the executing request, which affects the output of link generation. All of the <xref:Microsoft.AspNetCore.Routing.LinkGenerator> APIs allow specifying a base path. Always specify an empty base path to undo `Map*`'s affect on link generation.
 
+## Endpoint routing
+
+* A route endpoint has a template, metadata, and a request delegate that serves the endpoint's response. The metadata is used to implement cross-cutting concerns based on policies and configuration attached to each endpoint. For example, an Authorization Middleware can interrogate the endpoint's metadata collection for an `[authorization policy](xref:security/authorization/policies#applying-policies-to-mvc-controllers)`.
+* Routing is connected to the [middleware](xref:fundamentals/middleware/index) pipeline by the <xref:Microsoft.AspNetCore.Builder.RouterMiddleware> class. [ASP.NET Core MVC](xref:mvc/overview) adds routing to the middleware pipeline as part of its configuration and handles routing in MVC and Razor Pages apps. Endpoint data sources are mapped when MVC adds routing to the middleware pipeline. This routing middleware handles matching the template. At the end of the request pipeline a matched endpoint's request delegate is automatically called.
+* If the apps endpoints are static, endpoints can be mapped using <xref:Microsoft.AspNetCore.Builder.EndpointRouteBuilderExtensions.MapGet*>, <xref:Microsoft.AspNetCore.Builder.EndpointRouteBuilderExtensions.MapPost*>, and MapVerb helper methods. Static endpoints don't change during the lifetime of the app.
+* If endpoints are dynamic, a custom data source must be created. For example, a new Razor page is added to the app at runtime and needs a new endpoint.
```

Comment:
I think the terms static and dynamic might mislead people. Loading route configuration from a file or database at startup is what a lot of people would think of as dynamic, but these would be static routes because they don't change after the app has started.